boot/makebootable.go

```diff
@@ -487,7 +496,12 @@ func makeRunnableSystem(model *asserts.Model, bootWith *BootableSet, sealer *Tru
 		}
 		modeenv.CurrentKernelCommandLines = bootCommandLines{cmdline}
 
-		cmdlineVars, err := bootVarsForTrustedCommandLineFromGadget(bootWith.UnpackedGadgetDir)
+		// Look at gadget default values for system.kernel.*cmdline-append options
+		cmdlineAppend, err := buildOptionalKernelCommandLine(model, bootWith.UnpackedGadgetDir)
```

Comment:
This maybe a very silly question - so the way I read this that this will build a OR-policy that allows any of the `{,dangerous}-cmdline-append` options. So a user could press ESC during the boot and just append the cmdline option in this allow-list and the boot continues just fine, is that understanding correct? One use-case we have for this is that the customer uses "snap set" to allow/deny a special bootflag but it would be a privileged operation. Maybe (probably?) I misunderstand but the way I read this is that any user with access to the boot process could ESC and add the parameter?

Reply:
Hm, not really, what this does is building a command line with the current system.kernel.*cmdline-append options and then sets appropriately the grub environment so you get the expected kernel command line on boot. And the full command line is measured by the TPM, so it is not like you can select one or another argument from grub, you would need to change the options to whatever you need. So the scenario of a user adding a kernel argument from grub would not be possible.

Reply:
Silly me, indeed, I see my confusion now.